enterprise/web.jspparser/external/binaries-list

```diff
@@ -19,3 +19,6 @@
 F48473482C0E3E714F87186D9305BCAE30B7F5CB org.glassfish:jakarta.el:3.0.4
 9A3D933B7B079AD73C43878BF09AC47586FAC897 org.glassfish.main.common:glassfish-ee-api:5.1.0
 CED8804F81DB0C9099786472C615A12EA592C44B org.glassfish.main.common:common-util:5.1.0
+34A035507F0270F1C6B7722D728BD7B5A9BBAC4C jakarta.servlet.jsp.jstl:jakarta.servlet.jsp.jstl-api:1.2.7
+F5A092DE3B2B087C14CA4B8D6F2C77A1F6802828 org.glassfish.web:jakarta.servlet.jsp.jstl:1.2.6
+8CFF7DC1E492911F3DD7640EBFB60D6206A1DD40 commons-logging:commons-logging:1.3.1
```

Comment:
if this is only used for tests, why can't it copy the jars from the existing lib wrappers `enterprise/libs.jstl` and `platform/o.apache.commons.logging`?

This would avoid having to deal with license files again and would simplify maintenance over time.

Reply:
Because then I have an implicit dependency on these modules. We already saw breakage because binary at place $x was modified and a unittest and place $y then failed. That was not caught because CI/CD does not run all unittests. The licenses are trivial and there are no updates expected.

Reply:
I think the fact that it is for tests is a good argument that it should be simply copied from the wrapper, since worst what could happen is that CI catches it after it is merged. We also made some efforts to deduplicate and sync up all asm users which worked well.

IMO: lib wrapper update PRs for commonly used libs should always run all tests - so its caught right away. (e.g I ran all tests for flatlaf once https://github.com/apache/netbeans/pull/7565)

I would be still for reusing the shared libs, but if not: can you add a comment to the binaries-list file that this is test data and updates should be ignored?

Reply:
> We also made some efforts to deduplicate and sync up all asm users which worked well.

I'm a bit fuzzy, but if I remember correctly, in this case asm was used at runtime. That is a totally different case then the usage here for sample projects.

>IMO: lib wrapper update PRs for commonly used libs should always run all tests - so its caught right away. (e.g I ran all tests for flatlaf once https://github.com/apache/netbeans/pull/7565)

a) we have tests, that are partially deactivated
b) we have tests that don't run at all at CI/CD time

So no. It won't be caught. Case in point: This PR reenables unittests which bitrottet.

>can you add a comment to the binaries-list file that this is test data and updates should be ignored?

https://github.com/apache/netbeans/blob/c7e9bada8d0c41c30545962085e53ca38ad3414d/nbbuild/antsrc/org/netbeans/nbbuild/extlibs/ignored-overlaps#L125
https://github.com/apache/netbeans/blob/c7e9bada8d0c41c30545962085e53ca38ad3414d/enterprise/web.jspparser/external/jakarta.servlet.jsp.jstl-1.2-license.txt#L7
https://github.com/apache/netbeans/blob/c7e9bada8d0c41c30545962085e53ca38ad3414d/enterprise/web.jspparser/external/commons-logging-1.3.1-license.txt#L7

From my perspective this is a total non-issue. I have some hopes, that nobody run around blindly updating binaries without checking what the module does with these. If I remember correctly the jstl binaries are even from the JavaEE area and thus will never seen an update.

Reply:
> So no. It won't be caught. Case in point: This PR reenables unittests which bitrottet.

well yes. If tests were never active in the first place this can't be found by CI. But you fixed them with this PR.

>  I have some hopes, that nobody run around blindly updating binaries without checking what the module does with these.

So do I, since lib updates need to be tested (often manually). And lib update PRs should run https://github.com/apache/netbeans/labels/ci%3Aall-tests at least once as previously mentioned. Post-merge everything runs btw, the JDK/OS matrix is just reduced.

I did see the comment in the ignored-overlaps file, but this is not the first place where someone looks at when checking a version. It would be good to have a comment in the binaries-list file directly where it is declared.

I am working on a commons lib update PR right now actually, based on https://github.com/apache/netbeans/actions/runs/9912053909. This will show up too.

Reply:
Whatever. This just wastes my time, so if it makes you happy, I'll add a comment.

Reply:
wow

Reply:
Sorry, you hit the wrong nerve and the wrong time. Updated in place, please have another look.

Reply:
all good, bad days happen - apology accepted ;)